packages/edit-site/src/components/page-patterns/delete-category-menu-item.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	MenuItem,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { PATTERN_TYPES, PATTERN_DEFAULT_CATEGORY } from '../../utils/constants';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+export default function DeleteCategoryMenuItem( { category, onClose } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const history = useHistory();
+
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	const { deleteEntityRecord, invalidateResolution } =
+		useDispatch( coreStore );
+
+	if ( ! category?.id ) {
+		return null;
+	}
```

Comment:
This isn't quite as complex as the rename dialog, but I wonder if we still want to pull the modal and the logic into the patterns package?

Reply:
There are a few semi-related things underway at the moment that might get consolidated into, or update, the patterns package.  For example, [rename modal/command for patterns](https://github.com/WordPress/gutenberg/pull/55188) themselves, [alternate duplication flows](https://github.com/WordPress/gutenberg/pull/54764) etc.

I went in circles a bit before deciding to simply get a few proof of concepts up so it was a little clearer what could or should be extracted to the patterns package. 

There's also the fact that some code has already been refactored lately to combine components for templates, template parts, and patterns. That blurs the lines somewhat and makes consistently separating all patterns-related stuff out more difficult.

To begin with, adding new modal components to the patterns package seemed like the easiest line to draw. The `ConfirmDialog` fell on the other side of that line. 

I'd like to see how the different uses for the modals, e.g. menu items vs commands etc, shape up before we go bringing everything in. I'm certainly not opposed to that once we know more about our requirements. 

I also wonder if some of that refactoring should happen when we consolidate different types of patterns e.g. template parts 🤔

Reply:
That sounds like a good approach, thanks for outlining your thinking around it.